### PR TITLE
Add normalizeIconClass filter

### PR DESCRIPTION
**Pull request by rhamilto**
Supports https://github.com/openshift/origin/issues/17745

Required by https://github.com/openshift/origin-web-catalog/pull/603 and https://github.com/openshift/origin-web-console/pull/2607

### CODE REVIEW

File: src/filters/normalizeIconClass.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+angular.module("openshiftCommonUI")
+  .filter("normalizeIconClass", function() {
+    return function(iconClass) {
+      // if iconClass starts with "icon", append "icon-font "
+      // so the Openshift Logos Icon font is used
+      if(_.startsWith(iconClass, "icon")) {
```
Comment (spadgett) [inline src/filters/normalizeIconClass.js]: Should we be checking for "icon" or "icon-"? I believe all of the OpenShift logos start with "icon-"

File: src/filters/normalizeIconClass.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+angular.module("openshiftCommonUI")
+  .filter("normalizeIconClass", function() {
+    return function(iconClass) {
+      // if iconClass starts with "icon", append "icon-font "
```
Comment (spadgett) [inline src/filters/normalizeIconClass.js]: s/icon-font/font-icon

File: src/filters/normalizeIconClass.js (hunk 1 above)
Comment (spadgett) [reply src/filters/normalizeIconClass.js]: It might be good to keep the hyphen to avoid false positives

**Comment by rhamilto** [comment]
Good catches, @spadgett.  Fixed.

**Pull request merged**